components/molecule/drawer/src/index.js

```diff
@@ -0,0 +1,117 @@
+import {useState, useEffect, useRef, useCallback} from 'react'
+import PropTypes from 'prop-types'
+import {createPortal} from 'react-dom'
+import cx from 'classnames'
+
+const useOnCloseAnimation = isOpen => {
+  const [showDrawer, setShowDrawer] = useState(isOpen)
+  useEffect(() => {
+    isOpen && setShowDrawer(isOpen)
+  }, [isOpen])
+
+  const handlerOnAnimationEnd = () => {
+    !isOpen && setShowDrawer(false)
+  }
+
+  return [showDrawer, handlerOnAnimationEnd]
+}
+
+const Overlay = 'div'
+const Body = 'div'
+const Content = 'div'
+
+const PLACEMENTS = {
+  TOP: 'top',
+  RIGHT: 'right',
+  BOTTOM: 'bottom',
+  LEFT: 'left'
+}
+
+export default function MoleculeDrawer({
+  portalContainerId = 'drawer-react-portal',
+  isOpen = false,
+  placement = 'left',
+  onClose,
+  children
+}) {
+  const overlayRef = useRef(null)
+  const contentRef = useRef(null)
+  const [isClientReady, setClientReady] = useState(false)
+  const [isOpenState, handlerOnAnimationEnd] = useOnCloseAnimation(isOpen)
+
+  useEffect(() => {
+    setClientReady(true)
+  }, [])
+
+  const onKeyDown = useCallback(
+    ev => {
+      if (isOpen === false) return
+      if (ev.key === 'Escape') {
+        onClose(ev)
+        ev.preventDefault()
+      }
+    },
+    [isOpen, onClose]
+  )
+
+  useEffect(() => {
+    document.removeEventListener('keydown', onKeyDown)
```

Comment:
Is this truly needed ?

Reply:
I guess not, we are already cleaning up everything in the unmount callback